internal/featuregates/featuregates.go

```diff
@@ -2,43 +2,55 @@ package featuregates
 
 import (
 	"context"
+
 	corev1 "k8s.io/api/core/v1"
-	"log"
+	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type FeatureGates struct {
-	Client        client.Client
-	Namespace     string
-	ConfigMapName string
-}
+const (
+	TimeTravelFeatureGate = "timeTravel"
+	FgConfigMapName       = "osc-feature-gates"
+	OperatorNamespace     = "openshift-sandboxed-containers-operator"
+)
 
 var DefaultFeatureGates = map[string]bool{
-	"timeTravel":              false,
-	"quantumEntanglementSync": false,
-	"autoHealingWithAI":       true,
+	"timeTravel": false,
+}
+
+type FeatureGateStatus struct {
+	FeatureGates map[string]bool
 }
 
-func (fg *FeatureGates) IsEnabled(ctx context.Context, feature string) bool {
-	if fg == nil {
-		return false
+// This method returns a new FeatureGateStatus object
+// that contains the status of the feature gates
+// defined in the ConfigMap in the namespace
+// If there are any errors in retrieving the ConfigMap, or the feature gates
+// are not defined in the ConfigMap, the default values are used
+func NewFeatureGateStatus(client client.Client) (*FeatureGateStatus, error) {
+	fgStatus := &FeatureGateStatus{
+		FeatureGates: make(map[string]bool),
 	}
+
 	cfgMap := &corev1.ConfigMap{}
-	err := fg.Client.Get(ctx,
-		client.ObjectKey{Name: fg.ConfigMapName, Namespace: fg.Namespace},
-		cfgMap)
-
-	if err != nil {
-		log.Printf("Error fetching feature gates: %v", err)
-	} else {
-		if value, exists := cfgMap.Data[feature]; exists {
-			return value == "true"
+	err := client.Get(context.TODO(), types.NamespacedName{Name: FgConfigMapName,
+		Namespace: OperatorNamespace}, cfgMap)
+	if err == nil {
+		for feature, value := range cfgMap.Data {
+			fgStatus.FeatureGates[feature] = value == "true"
 		}
 	}
 
-	defaultValue, exists := DefaultFeatureGates[feature]
-	if exists {
-		return defaultValue
+	// Add default values for missing feature gates
+	for feature, defaultValue := range DefaultFeatureGates {
+		if _, exists := fgStatus.FeatureGates[feature]; !exists {
+			fgStatus.FeatureGates[feature] = defaultValue
+		}
 	}
-	return false
+	return fgStatus, err
+}
+
+func IsEnabled(fgStatus *FeatureGateStatus, feature string) bool {
+
+	return fgStatus.FeatureGates[feature]
```

Comment:
Since feature is a string, and anything could be used, I'm missing here a KeyError handler.

Reply:
@beraldoleal can you please share an example..

Reply:
Sorry, I'm unable to understand this and hence my question. The FG status is populated during every reconcile and checked by the operator code so the code has to ensure right feature is checked.
Also if a missing feature is checked it'll return the default bool value, ie false. 
Check this - thttps://play.golang.com/p/X2N0sc5fNi6

Reply:
Right, maybe I misinterpreted Beraldo's concern.  @beraldoleal could you elaborate what you're worried about?

Reply:
Frankly, it'd probably leave it as is.  First, gated feature identifiers are very likely to stay the way they are - compile-time constants controlled by this project.  In other words, I don't see much scope for them to become somewhat arbitrary strings.  Also, it could be argued that if a feature doesn't exist then it's necessarily disabled :-) so even the worst case semantics doesn't really disturb me.

Reply:
Note that the caller (ie the operator code) needs to explicitly query to check a feature gate. So why would we willing write a code for non-existent feature. If there are cases about using featuregate module elsewhere then it might make sense. Let's handle it when there is a requirement.

Reply:
Yes, I think it's robust.

Reply:
> So why would we willingly write a code for non-existent feature?

> First, gated feature identifiers are very likely to stay the way they are - compile-time constants controlled by this project. In other words, I don't see much scope for them to become somewhat arbitrary strings. 

We can't guarantee future code, refactoring, or features won't break this assumption. Writing code based on an assumption without enforcement is risky. However, I agree we can't account for every possibility and need to find a balance. Over-engineering is bad, but lacking some checks is also dangerous.

So, for the sake of balance, let's move forward. Approved.